templates/utils.js

```diff
@@ -2,45 +2,195 @@ import { inspect } from "util";
 import createDeepMerge from "@fastify/deepmerge";
 
 // https://github.com/fastify/deepmerge?tab=readme-ov-file#mergearray Example 1
-const replaceByClonedSource = (options) => {
-  const clone = options.clone
-  return  (_target, source) => {
-    return clone(source)
-  }
-}
+const replaceByClonedSource = options => {
+  const clone = options.clone;
+  return (_target, source) => {
+    return clone(source);
+  };
+};
+
+const deepMergeWithoutKeysOrder = createDeepMerge({ mergeArray: replaceByClonedSource });
 
-export const deepMerge = createDeepMerge({ mergeArray: replaceByClonedSource });
+export const deepMerge = (...args) => {
+  const mergedConfig = deepMergeWithoutKeysOrder(...args);
+  const finalConfig = {};
+  for (const key of Object.keys(args[0])) {
+    finalConfig[key] = mergedConfig[key];
+  }
+  for (const key of Object.keys(mergedConfig)) {
+    if (!(key in finalConfig)) {
+      finalConfig[key] = mergedConfig[key];
+    }
+  }
+  return finalConfig;
+};
 
 export const withDefaults =
   (template, expectedArgsDefaults, debug = false) =>
-  (receivedArgs) => {
+  receivedArgs => {
     const argsWithDefault = Object.fromEntries(
-      Object.entries(expectedArgsDefaults)
-      .map(([argName, argDefault]) => [argName, receivedArgs[argName] ?? [argDefault]])
+      Object.entries(expectedArgsDefaults).map(([argName, argDefault]) => [
+        argName,
+        receivedArgs[argName] ?? [argDefault],
+      ]),
     );
 
     if (debug) {
       console.log(argsWithDefault, expectedArgsDefaults, receivedArgs);
     }
 
-    const expectedArgsNames = Object.keys(expectedArgsDefaults)
-    Object.keys(receivedArgs).forEach((receivedArgName) => {
+    const expectedArgsNames = Object.keys(expectedArgsDefaults);
+    Object.keys(receivedArgs).forEach(receivedArgName => {
       if (!expectedArgsNames.includes(receivedArgName)) {
         throw new Error(
-          `Templated received unexpected argument \`${receivedArgName}\`. Expecting only ${
-            expectedArgsNames.map(name => `\`${name}\``).join(", ")
-          }`
+          `Templated received unexpected argument \`${receivedArgName}\`. Expecting only ${expectedArgsNames
+            .map(name => `\`${name}\``)
+            .join(", ")}`,
         );
       }
     });
 
     return template(argsWithDefault);
   };
 
-export const stringify = val => {
-  const str = inspect(val, { depth: null, compact: true, maxArrayLength: null, maxStringLength: null });
+const addComments = (val, comments = {}) => {
+  let str = inspect(val, {
+    depth: null,
+    compact: false,
+    maxArrayLength: null,
+    maxStringLength: null,
+  });
+
+  // Add comments above their respective properties
+  if (Object.keys(comments).length > 0) {
```

Comment:
Maybe we do early return? 

```
if (Object.key(comments).length === 0) return str;
``` 

So the the code easy parse?

Reply:
thanks, updated [b9bc75a](https://github.com/scaffold-eth/create-eth/pull/227/commits/b9bc75a14f2858e6fe6d196221d3cf4e81c9392b)